store/sqlstore/status_store.go

```diff
@@ -98,6 +101,119 @@ func (s SqlStatusStore) GetByIds(userIds []string) ([]*model.Status, error) {
 	return statuses, nil
 }
 
+// MySQL doesn't have support for RETURNING clause, so we use a transaction to get the updated rows.
+func (s SqlStatusStore) updateExpiredStatuses(t *gorp.Transaction) ([]*model.Status, error) {
+	var statuses []*model.Status
+	currUnixTime := time.Now().UTC().Unix()
+	selectQuery, selectParams, err := s.getQueryBuilder().
+		Select("*").
+		From("Status").
+		Where(
+			sq.And{
+				sq.Eq{"Status": model.STATUS_DND},
+				sq.Gt{"DNDEndTime": 0},
+				sq.LtOrEq{"DNDEndTime": currUnixTime},
+			},
+		).ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "status_tosql")
+	}
+	_, err = t.Select(&statuses, selectQuery, selectParams...)
```

Comment:
Looks like you are selecting the "old" rows, not the "updated ones", is it what is expected?

Reply:
ok, you are updating them by hand later outside the function, looks like there is some kind of displacement of resposability, for me, the update of the values returning the right statuses should be encapsulated in the same function where the select and the update done.

Reply:
@jespino agniva suggested this approach to avoid making extra db call. open to get it from db if needed.